romanisim/catalog.py

```diff
@@ -241,6 +442,73 @@ def make_galaxies(coord,
         # sigma of one mag isn't nuts.  But this will be totally uncorrelated
         # in different bands, so we'll get some weird colored objects
         out[bandpass] = (10.**(-mag_thisband / 2.5)).astype('f4')
+
+    return out
+
+
+def make_gaia_stars(coord,
+                    radius=0.1,
+                    date=None,
+                    bandpasses=None,
+                    rng=None,
+                    seed=50,
+                    **kwargs
+                    ):
+    """Make a catalog of stars from the GAIA catalog.
+
+    Parameters
+    ----------
+    coord : astropy.coordinates.SkyCoord
+        Location around which to generate sources.
+    radius : float
+        Radius in degrees in which to generate sources
+    date : astropy.time.Time
+        Optional argument to provide a date and time for stellar search
+    bandpasses : list[str]
+        List of names of bandpasses for which to generate fluxes.
+    rng : galsim.BaseDeviate
+        Random number generator to use.
+    seed : int
+        Seed for random number generator to use, only used if rng is None.
+
+    Returns
+    -------
+    catalog : astropy.Table
+        Table for use with table_to_catalog to generate catalog for simulation.
+    """
+
+    if bandpasses is None:
+        bandpasses = ["F062", "F087", "F106", "F129", "F146", "F158", "F184", "F213"]
+
+    if date is None:
+        date = astropy.time.Time('2026-01-01T00:00:00')
+
+    if rng is None:
+        rng = galsim.UniformDeviate(seed)
+
+    # Perform GAIA search
+    q = f'select * from gaiadr3.gaia_source where distance({coord.ra.value}, {coord.dec.value}, ra, dec) < {radius}'
+    job = Gaia.launch_job_async(q)
+    r = job.get_results()
+
+    # Create catalog
+    star_cat = rsim_gaia.gaia2romanisimcat(r, date, fluxfields=bandpasses)
+
+    # Set object types
+    types = np.zeros(len(star_cat), dtype='U3')
+    types[:] = 'PSF'
+
+    # Populate output table
+    out = table.Table()
+    out['ra'] = star_cat['ra']
+    out['dec'] = star_cat['dec']
+    out['type'] = types
+    out['n'] = -1 * np.zeros(len(star_cat), dtype='f4')
+    out['half_light_radius'] = np.zeros(len(star_cat), dtype='f4')
+    out['pa'] = np.zeros(len(star_cat), dtype='f4')
+    out['ba'] = np.ones(len(star_cat), dtype='f4')
+    for bandpass in bandpasses:
+        out[bandpass] = star_cat[bandpass]
```

Comment:
What is this function doing on top of returning star_cat?  I'm missing the difference between out and star_cat.

The docs reference "false color" stars but gaia2romanisimcat just assumes color = 0 in all possible bands.  I briefly thought when you said that you might have meant that you were using Gaia BP/RP to predict the Roman bands, but that's not what's happening.  I've made a suggestion for the docs there.

Reply:
I did not realize that star_cat was already in the correct format. I will streamline this.

Reply:
Thanks.  The first part of this function looks reasonable but it duplicates
https://github.com/spacetelescope/romanisim/blob/main/scripts/romanisim-gaia-catalog#L28-L33
It's fine to have it here instead but then let's refactor romanisim-gaia-catalog to use it.  Looking more at this, it looks like both of us ended up writing romanisim-gaia-catalog / romanisim-make-catalog methods.  Can you make sure that both romanisim-make-catalog and romanisim-gaia-catalog do the same thing and then merge / get rid of one as part of this?  It looks like we made mildly different decisions about what arguments are required vs. not but the two modules are pretty identical.

Reply:
Addressed (both the cat question and the docs update).

Reply:
I streamlined my GAIA bit to nto have the catalog duplication. The function needs to have the code from the script, as the script is not important - however the script could be changed to use the catalog function. 

Do you want me to update the script in this ticket or a new one?

Reply:
Your call; I just want to eliminate the code duplication.